packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/common.ts

```diff
@@ -146,6 +150,28 @@ export function getCommonConfig(wco: WebpackConfigOptions): Configuration {
           // Add zone.js legacy support to the es5 polyfills
           // This is a noop execution-wise if zone-evergreen is not used.
           entryPoints[polyfillsChunkName].push('zone.js/dist/zone-legacy');
+
+          // Since the chunkFileName option schema does not allow the function overload, add a plugin
+          // that changes the name of the ES5 polyfills chunk to not include ES2015.
+          extraPlugins.push({
+            apply(compiler) {
+              compiler.hooks.compilation.tap('build-angular', compilation => {
+                // Webpack typings do not contain MainTemplate assetPath hook
+                // The webpack.Compilation assetPath hook is a noop in 4.x so the template must be used
+                // tslint:disable-next-line: no-any
+                (compilation.mainTemplate.hooks as any).assetPath.tap(
+                  'build-angular',
+                  (filename: string, data: ChunkData) => {
+                    const isMap = filename && filename.endsWith('.map');
+
+                    return data.chunk && data.chunk.name === 'polyfills-es5'
+                      ? `polyfills-es5${hashFormat.chunk}.js${isMap ? '.map' : ''}`
```

Comment:
I'm a bit confused here, this doesn't seem to change the name of `polyfills-es5-es2015.js`, instead it seems to target `polyfills-es5.js`.

Reply:
The `assetPath` hook is used by webpack to create the output filename/path for each chunk.   The filename that is passed as an argument to the hook is actually the filename template from the webpack configuration (defined [here](https://github.com/angular/angular-cli/blob/master/packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/common.ts#L477) in the CLI).  Chunk names are defined via the entry point configuration (`polyfills-es5` is defined [here](https://github.com/angular/angular-cli/blob/master/packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/common.ts#L144)).
This plugin allows the template for the output filename to be altered on a per chunk basis.  In this case, the `polyfills-es5` chunk's filename template is changed to remove the `targetInFileName` segment (`-es2015` in the differential loading case).  This causes webpack to generate the correct name and removes the need to patch the stats file and clean up the filenames after the fact.

Reply:
Ok I see now, thank you. The logic is a bit circuitous and I worry about someone else altering this code. Can you add it as a comment please?